lib/mohawk/navigation.rb

```diff
@@ -4,11 +4,13 @@ module Mohawk
   module Navigation
     include PageNavigation
 
-    def on(cls, extra={}, &block)
+    def on(cls, extra = {}, &block)
       screen = cls.new extra
       screen.wait_until_present
       block.call screen if block
       screen
+    rescue Mohawk::Waiter::WaitTimeout => e
+      raise e.class, "Unable to locate '#{cls}'", e.backtrace

```

Comment:
I wonder if there would be benefit to augmenting `Waiter` to accept a "context" on what you're waiting on, and pass that along down through everything else?

Reply:
Would the "context" be like an optional parameters that you pass to the `wait_until` call? I could get on board with that.
I was trying to figure out a way to make the Waiter error reporting better across the board but I only came up with a solution for class initialization.

Reply:
Yeah, I was thinking something like making the signature `def wait_until(timeout=Mohawk.timeout, context=nil, &block)` or something like that and use the provided value when it does the explicit `raise` (so you'd have to opt-into it).